File: Proyecto Java/SoftCapsuleCare-Users/src/pe/edu/pucp/citamedica/model/usuario/Usuario.java
package pe.edu.pucp.citamedica.model.usuario;
import pe.edu.pucp.citamedica.model.usuario.Persona;

public class Usuario {
    private static int contador = 0;

    private int idUsuario;
    private String username;
    private String contrasenha;
    private boolean activo;
    private int idPersona;

    public Usuario() {
    }

    public Usuario(int idUsuario, String username, String contrasenha, boolean activo, int idPersona) {
        this.idUsuario = ++contador;
        this.username = username;
        this.contrasenha = contrasenha;
        this.activo = activo;
        this.idPersona = idPersona;
    }

    public int getIdPersona() {
        return idPersona;
    }

    public void setIdPersona(int idPersona) {
        this.idPersona = idPersona;
    }
    public int getIdUsuario() {
        return idUsuario;
    }

    public void setIdUsuario(int idUsuario) {
        this.idUsuario = idUsuario;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }
    
    public String getContrasenha() {
        return contrasenha;
    }

    public void setContrasenha(String contrasenha) {
        this.contrasenha = contrasenha;
    }
    
    public boolean isActivo() {
        return activo;
    }

    public void setActivo(boolean activo) {
        this.activo = activo;
    }

    public void actualizarInformacion() {
        // Implementar lógica de actualización
    }

    @Override
    public String toString() {
        return "Usuario{" + "idUsuario=" + idUsuario + ", username=" + username + ", contrasenha=" + contrasenha + ", idPersona=" + idPersona + '}';
    }
    

}

File: ClinicaManagement/ClinicaManagement/ClinicaModel/Usuario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClinicaModel
{
    public class Usuario
    {
        private static int contador = 0;
        private int idUsuario;
        private string username;
        private string contrasenha;
        private bool activo;
        private int idPersona;

        public int IdUsuario { get; set; }
        public string Username { get; set; }
        public string Contrasenha { get; set; }
        public bool Activo { get; set; }
        public int IdPersona { get; set; }

        public Usuario()
        {
            IdUsuario = ++contador;
        }

        public Usuario(string username, string contrasenha, bool activo, int idPersona)
        {
            IdUsuario = ++contador;
            Username = username;
            Contrasenha = contrasenha;
            Activo = activo;
            IdPersona = idPersona;
        }
    }
}
